### ISSUE TITLE
Caldav integration is creating events in every calendars

### ISSUE THREAD

**Issue by PicchiKevin** [calcom/cal.com#537]
### Issue Summary
When using the CalDav integration and having multiple calendar on the CalDav server, the event is created in all the calendars.

Here is the outcome in the calendar:
![image](https://user-images.githubusercontent.com/4269168/131337095-111e23c0-0dc2-411d-ba50-69e0df1a82b2.png)

### Steps to Reproduce
1. Connect to your CalDav server
2. Create an event
3. Observe the result in your synced calendar.

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

The event should only be inserted in either the main calendar or allow the user to select a calendar where calendso event will be added.

The code responsible for the bug is laying in the adapter: https://github.com/calendso/calendso/blob/8e9703545af208aa073783a9f0643e635943e035/lib/integrations/CalDav/CalDavCalendarAdapter.ts#L96-L107

**Comment by bennetfabian** [comment]
Good catch. Already observed this myself but now that you already created this issue I don't need to do it myself. Thank you very much for reporting.

**Comment by PicchiKevin** [comment]
As a user, I'd prefer option 1.

Not sure if that what you mean but considering the current architecture, I'd keep the current flow but would only add a dropdown list in the "CalDav Server App" Integration details. And if the user hasn't selected anything yet, well it would just default to the first calendar.

What do you think? 

(BTW it would be great to have more information about the current connection in the integration details)

**Comment by bennetfabian** [comment]
> As a user, I'd prefer option 1.
> 
> Not sure if that what you mean but considering the current architecture, I'd keep the current flow but would only add a dropdown list in the "CalDav Server App" Integration details. And if the user hasn't selected anything yet, well it would just default to the first calendar.
> 
> What do you think?
> 
> (BTW it would be great to have more information about the current connection in the integration details)

For the first part, I agree with you, it could definitely be solved by having a dropdown (multi-option perhaps) in the CalDav integration details even though it may be hard to find for some. If we had the dropdown in the App Store main page like the "Select calendars" (for availability) modal, it could be found easier. (even though it may be specific to CalDav)
I think there should be further discussion about this topic. **@ all maintainers**

For the maximum of configurability, I thought we may want to have the ability to override the user's global choice of which calendars the event gets created in on a 'per event type' basis.

Maybe Peer, Alex, Bailey, as well as all the other maintainers reading this could contribute their opinions about how this topic should be implemented. At least it's pretty clear that we need to change the current behavior of creating an event in each and every calendar.

**Comment by PeerRich** [comment]
adding @femyeda who worked on CalDav

**Comment by aspiers** [comment]
Is this issue really unique to Caldav as the issue title suggests?  I have connected two Google accounts and nothing else, and when I created a test invite, the same event got created in both accounts, which means that my unified calendar shows the same event twice in two different colours, one from each account.

I am a refugee from x.ai (a great service, but proprietary and recently got effectively killed in a bizarre acquisition by Bizzabo), and they had a simple and effective solution for this: for each event type the user sets up, allow the user to specify which calendar in which the new event will be created.

**Comment by joshtemple** [comment]
I have the same issue for Google Calendar. I've connected 3 accounts and new events are created on all three. This is undesirable for me as I already have external functionality to sync events the way I want them. Instead, I'd like to be able to configure where events get created.

At a minimum, a global setting, though being able to do it by meeting type would be even better (a meeting I can share for events where I want them booked to my personal calendar, for example).

**Comment by joshtemple** [comment]
@PeerRich I wanted to point out that this issue does not seem to be limited to CalDav as myself (Google Calendar) and @aspiers reported above. It may be worth updating the issue title accordingly.

**Comment by juicejs** [comment]
Are there any updates on this? We are using CalDAV as a team with delegated calendars. So we have multiple calendars representing different people. It actually makes Calendso useless in our team environment, because events are not getting added per person but for the whole team.

**Comment by philprime** [comment]
I would like to ping this issue, as I experienced the same issue as @PicchiKevin with my Apple Calendar.

It created the exact same event in all my iCloud calendars (noticed it because it created new entries in the ones I share with my friends/family).

**Comment by PeerRich** [comment]
> I would like to ping this issue, as I experienced the same issue as @PicchiKevin with my Apple Calendar.
> 
> It created the exact same event in all my iCloud calendars (noticed it because it created new entries in the ones I share with my friends/family).

@natelindev can you take a look into this?

**Comment by natelindev** [comment]
I believe this issue have already been solved by the option here.
You can pick which calendar you want to create events on.

<img width="332" alt="Screen Shot 2022-01-10 at 10 39 10 PM" src="https://user-images.githubusercontent.com/9290574/148783982-3db9f107-65b8-4a38-ae4f-4443cdb91279.png">

**Comment by philprime** [comment]
Yes the option exist, but it seems broken.

I am using the latest `main` branch and set up the following integrations:

<img width="1230" alt="Screenshot 2022-01-10 at 21 14 12" src="https://user-images.githubusercontent.com/30923007/148834814-b18abdd3-0cc0-4e41-8530-3e9991081222.png">

As you can see I create a calendar called "Calendso Bookings" in Google Calendar, which is also selected as the target calendar for bookings.

I added my personal Apple Calendar so personal events are reducing my available time slots.

When a new booking is created, it tries to create it fails to create the booking in Google.


```
20:20:59.865 timeZoneName ERROR
 Error  Error creating event: , , , , , , ,
error stack:
• 6222.js:334 createEvent
    .next/server/chunks/6222.js:334:23

• 6222.js:140 createEvent
    .next/server/chunks/6222.js:140:39

• 8113.js:219 createAllCalendarEvents
    .next/server/chunks/8113.js:219:13

• 8113.js:103 create
    .next/server/chunks/8113.js:103:25

• confirm.js:335 handler
    .next/server/pages/api/book/confirm.js:335:36

• api-utils.ts:107 apiResolver
    node_modules/next/server/api-utils.ts:107:5

• next-server.ts:1515 handleApiRequest
    node_modules/next/server/next-server.ts:1515:5

• next-server.ts:1364 fn
    node_modules/next/server/next-server.ts:1364:27


20:20:59.915 timeZoneName ERROR CalendarManager createEvent failed
 Error  Error creating event: , , , , , , ,
error stack:
• 6222.js:334 createEvent
    .next/server/chunks/6222.js:334:23

• 6222.js:140 createEvent
    .next/server/chunks/6222.js:140:39

• 8113.js:219 createAllCalendarEvents
    .next/server/chunks/8113.js:219:13

• 8113.js:103 create
    .next/server/chunks/8113.js:103:25

• confirm.js:335 handler
    .next/server/pages/api/book/confirm.js:335:36

• api-utils.ts:107 apiResolver
    node_modules/next/server/api-utils.ts:107:5

• next-server.ts:1515 handleApiRequest
    node_modules/next/server/next-server.ts:1515:5

• next-server.ts:1364 fn
    node_modules/next/server/next-server.ts:1364:27


{
  type: '30 Min Meeting zwischen Philip Niedertscheider und Phil',
  title: '30 Min Meeting zwischen Philip Niedertscheider und Phil',
  description: '',
  startTime: '2022-01-12T12:00:00.000Z',
  endTime: '2022-01-12T12:30:00.000Z',
  organizer: {
    email: 'phil@kula.app',
    name: 'Philip Niedertscheider',
    timeZone: 'Europe/Amsterdam'
  },
  attendees: [
    {
      id: 9,
      email: 'phil@techprimate.com',
      name: 'Phil',
      timeZone: 'Europe/Vienna',
      bookingId: 9
    }
  ],
  location: 'integrations:google:meet',
  uid: '6VqtSoUAKB7UTcmNjykEwJ',
  language: [Function: fixedT] {
    lng: 'de',
    ns: 'common',
    keyPrefix: undefined
  },
  conferenceData: {
    createRequest: {
      requestId: '8a0ce172-f69a-51b5-8481-93feb73d58e2'
    }
  }
}
20:20:59.920 timeZoneName ERROR [api] book:user Booking phil failed
{
  errorCode: 'BookingCreatingMeetingFailed',
  message: 'Booking failed'
}
[
  {
    type: 'apple_calendar',
    success: false,
    uid: '6VqtSoUAKB7UTcmNjykEwJ',
    createdEvent: undefined,
    originalEvent: {
      type: '30 Min Meeting zwischen Philip Niedertscheider und Phil',
      title: '30 Min Meeting zwischen Philip Niedertscheider und Phil',
      description: '',
      startTime: '2022-01-12T12:00:00.000Z',
      endTime: '2022-01-12T12:30:00.000Z',
      organizer: {
        email: 'phil@kula.app',
        name: 'Philip Niedertscheider',
        timeZone: 'Europe/Amsterdam'
      },
      attendees: [
        {
          id: 9,
          email: 'phil@techprimate.com',
          name: 'Phil',
          timeZone: 'Europe/Vienna',
          bookingId: 9
        }
      ],
      location: 'integrations:google:meet',
      uid: '6VqtSoUAKB7UTcmNjykEwJ',
      language: [Function: fixedT] {
        lng: 'de',
        ns: 'common',
        keyPrefix: undefined
      },
      conferenceData: {
        createRequest: {
          requestId: '8a0ce172-f69a-51b5-8481-93feb73d58e2'
        }
      }
    }
  }
]
```

And creates duplicate entries in the Apple Calendar which should not happen in the first place.

<img width="1138" alt="Screenshot 2022-01-10 at 21 22 42" src="https://user-images.githubusercontent.com/30923007/148835022-d496c00f-199e-4060-a430-36ff4f759c7a.png">

I tried to look at the source code but it seems like I do not know enough about the credentials/calendar manager to understand how it works.

**Comment by philprime** [comment]
(wild guess) Why is the BaseCalendarService creating the event in every single calendar, even tough we can select a specific one?

https://github.com/calendso/calendso/blob/09c4040ce5da86e97dd6de08246f69876214e02c/lib/integrations/calendar/services/BaseCalendarService.ts#L53-L117

https://github.com/calendso/calendso/blob/09c4040ce5da86e97dd6de08246f69876214e02c/lib/integrations/calendar/services/BaseCalendarService.ts#L200-L227

**Comment by natelindev** [comment]
I'll investigate further today.

**Comment by joeauyeung** [comment]
Fixed in #3197
